sensor/temp-nct75.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Phoenix-RTOS
+ *
+ * NCT75 i2c Temperature Sensor driver
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Marek Bialowas
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+#include <sys/msg.h>
+#include <posix/utils.h>
+
+#include <i2c.h>
+
+
+static const uint32_t INVALID_TEMP = -1000 * 1000u;
+
+static const uint8_t DEV_ADDR = 0x48u;      /* NCT75 */
+static const uint8_t TEMP_REG_ADDR = 0x00u; /* stored temperature value - 16bit */
+
+
+/* returns temperature in miliCelsius */
+static int32_t getTemp(void)
+{
+	uint8_t temp_raw[2];
+	int16_t temp_enc;
+
+	if (i2c_regRead(DEV_ADDR, TEMP_REG_ADDR, temp_raw, sizeof(temp_raw)) < 0)
+		return INVALID_TEMP;
+
+	/* 12-bit left-adjusted, 2s complement, every bit is 0.0625 C */
+	temp_enc = (temp_raw[0] << 8) | temp_raw[1];
+	return (temp_enc >> 5) * 125;
```

Comment:
Hmm, if it is 12 bit left-adjusted, shouldn't it be here `(temp_enc >> 4)`?

Reply:
yeah, the code is sneaky here:
```c
(temp_enc >> 4) * 62.5 ~= (temp_enc >> 4) / 2 * 125 = (temp_enc >> 5) * 125
```
loosing one bit of information for the sake of not dealing with floats :)